app/jobs/appointment_notification/worker.rb

```diff
@@ -102,6 +61,42 @@ def send_message(notification, communication_type, recipient_number:)
     response
   end
 
+  def notify_via_twilio(notification, communication_type, recipient_number, context)
+    service = TwilioApiService.new(sms_sender: medication_reminder_sms_sender)
+    args = {
+      recipient_number: recipient_number,
+      message: notification.localized_message,
+      callback_url: callback_url,
+      context: context
+    }
+
+    # remove missed_visit_whatsapp_reminder and missed_visit_sms_reminder
+    # https://app.clubhouse.io/simpledotorg/story/3585/backfill-notifications-from-communications
+    handle_twilio_error(notification) do
+      case communication_type
+      when "whatsapp", "missed_visit_whatsapp_reminder"
+        service.send_whatsapp(args)
+      when "sms", "missed_visit_sms_reminder"
+        service.send_sms(args)
+      else
+        raise UnknownCommunicationType, "#{self.class.name} is not configured to handle communication type #{communication_type}"
+      end
+    end
+  end
+
+  # This usually happens when the phone number is invalid.
+  # https://www.twilio.com/docs/errors/21211
+  def handle_twilio_error(notification, &block)
+    block.call
+  rescue TwilioApiService::Error => error
```

Comment:
Would ideally like to handle this as data, not an exception, so that it's more first-class, not flow-by-exception, etc. But it's not important to address immediately. I would expect the gem to let us configure this behaviour, but I haven't looked into it deeply.